examples/dishwasher-app/silabs/src/DeviceEnergyManager.cpp

```diff
@@ -0,0 +1,167 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AppConfig.h"
+#include "AppTask.h"
+
+#include "DataModelHelper.h"
+#include "DeviceEnergyManagementDelegateImpl.h"
+#include "DeviceEnergyManager.h"
+#include "DishwasherManager.h"
+#include "EnergyTimeUtils.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::DeviceEnergyManagement;
+
+DeviceEnergyManager DeviceEnergyManager::sDeviceEnergy;
+
+namespace {
+std::unique_ptr<DeviceEnergyManagementDelegate> gDEMDelegate;
+std::unique_ptr<DeviceEnergyManagementManager> gDEMInstance;
+std::unique_ptr<DishwasherManager> gDishwasherManager;
+} // namespace
+
+DeviceEnergyManagement::DeviceEnergyManagementDelegate * GetDEMDelegate()
+{
+    VerifyOrDieWithMsg(gDEMDelegate.get() != nullptr, AppServer, "DEM Delegate is null");
+    return gDEMDelegate.get();
+}
+
+DishwasherManager * DeviceEnergyManagement::GetDishwasherManager()
+{
+    VerifyOrDieWithMsg(gDEMDelegate.get() != nullptr, AppServer, "Dishwasher Manager is null");
+    return gDishwasherManager.get();
+}
+
+/*
+ *  @brief  Creates a Delegate and Instance for DEM
+ *
+ * The Instance is a container around the Delegate, so
+ * create the Delegate first, then wrap it in the Instance
+ * Then call the Instance->Init() to register the attribute and command handlers
+ */
+CHIP_ERROR DeviceEnergyManagementInit()
+{
+    EndpointId DEMEndpointId = DataModelHelper::GetEndpointIdFromCluster(DeviceEnergyManagement::Id);
+    CHIP_ERROR err;
+
+    VerifyOrReturnError((DEMEndpointId != kInvalidEndpointId), CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(AppServer, "DEM Cluster not configured"));
+
+    // Initialize DEM (Device Energy Management) and DEMManufacturer (Device Energy Management Manufacturer)
+    VerifyOrReturnError(!gDEMDelegate && !gDEMInstance, CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(AppServer, "DEM Delegate or Instance already exist"));
+
+    gDEMDelegate = std::make_unique<DeviceEnergyManagementDelegate>();
+    VerifyOrReturnError(gDEMDelegate, CHIP_ERROR_NO_MEMORY, ChipLogError(AppServer, "Failed to allocate memory for DEM Delegate"));
+
+    // Manufacturer may optionally not support all features, commands & attributes
+    // DeviceEnergyManagement::Feature::kStateForecastReporting is removed from the specification [!PA].a,(STA|PAU|FA|CON),O
+    static uint32_t featureMap = static_cast<uint32_t>(DeviceEnergyManagement::Feature::kPowerForecastReporting) +
+        static_cast<uint32_t>(DeviceEnergyManagement::Feature::kStartTimeAdjustment) +
```

Comment:
I can't see where you are supporting Start Time Adjustment and Power Forecast Adjustment in the code - unless you are reusing the test examples.

It would be nice to put together an example dishwasher power forecast with representative real numbers.

But suspect we need a separate PR for that.

Reply:
That will also be put in the back log about improving this app, we are currently re-using the test examples for those just so we can pass matter certs.